File: RandomnessLibraries/DotNet/RandomnessDistributionsUnitTest/TestBaseClass.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NumpyDotNet;
using System;


namespace RandomnessDistributionsUnitTest
{
    public class TestBaseClass
    {
        [ClassInitialize]
        public static void CommonInit(TestContext t)
        {
            //  numpy.InitializeNumpyLibrary();
        }

        [TestInitialize]
        public void FunctionInit()
        {
        }

        protected void print(object obj)
        {
            if (obj == null)
            {
                Console.WriteLine("<null>");
                return;
            }
            if (obj is Array)
            {
                foreach (var o in (Array)obj)
                {
                    Console.Write(o.ToString() + ",");
                }
                Console.WriteLine("");
            }
            else
            {
                Console.WriteLine(obj.ToString());
            }
        }
        protected void print(string label, object obj)
        {
            Console.WriteLine(label + obj.ToString());
        }
        //protected void print<T>(string label, T[] array)
        //{
        //    Console.WriteLine(label + ArrayToString(array));
        //}


        internal void AssertArray<T>(T[] arrayData, T[] expectedData)
        {
            int length = expectedData.Length;

            for (int i = 0; i < length; i++)
            {
                T E1 = expectedData[i];
                T A1 = (T)arrayData[i];

                Assert.AreEqual(E1, A1);
            }
        }
        internal void AssertArray(double[] arrayData, double[] expectedData)
        {
            int length = expectedData.Length;

            for (int i = 0; i < length; i++)
            {
                double E1 = expectedData[i];
                double A1 = (double)arrayData[i];

                if (double.IsNaN(E1) && double.IsNaN(A1))
                    continue;
                if (double.IsInfinity(E1) && double.IsInfinity(A1))
                    continue;

                Assert.AreEqual(E1, A1, 0.00000001);
            }
        }

        internal Int32[] arange_Int32(int start, int end)
        {
            Int32[] buffer = new int[end - start];

            for (int i = 0; i < buffer.Length; i++)
            {
                buffer[i] = i + start;
            }

            return buffer;
        }

        internal double[] arange_Double(int start, int end)
        {
            double[] buffer = new double[end - start];

            for (int i = 0; i < buffer.Length; i++)
            {
                buffer[i] = (double)(i + start);
            }

            return buffer;
        }

        internal bool[] CompareArray(long[] arr, int c)
        {
            bool[] output = new bool[arr.Length];

            for (int i = 0; i < arr.Length; i++)
            {
                output[i] = arr[i] == c;
            }
            return output;
        }


        internal int CountTrues(bool[] v)
        {
            int count = 0;

            foreach (var b in v)
            {
                count += b == true ? 1 : 0;
            }

            return count;
        }

        internal double GetMax(double[] arr)
        {
            double highNum = double.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++)
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal long GetMax(long[] arr)
        {
            long highNum = long.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++)
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal ulong GetMax(ulong[] arr)
        {
            ulong highNum = ulong.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal int GetMax(int[] arr)
        {
            int highNum = int.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal uint GetMax(uint[] arr)
        {
            uint highNum = uint.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal short GetMax(short[] arr)
        {
            short highNum = short.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal ushort GetMax(ushort[] arr)
        {
            ushort highNum = ushort.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal sbyte GetMax(sbyte[] arr)
        {
            sbyte highNum = sbyte.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal byte GetMax(byte[] arr)
        {
            byte highNum = byte.MinValue;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] > highNum)
                    highNum = arr[i];
            }
            return highNum;
        }
        internal bool GetMax(bool[] arr)
        {
            bool highNum = false;
            int i;

            for (i = 0; i < arr.Length; i++) 
            {
                if (arr[i] == true)
                    highNum = arr[i];
            }
            return highNum;
        }

        internal double GetMin(double[] arr)
        {
            return (double)np.min(arr).GetItem(0);
        }
        internal long GetMin(long[] arr)
        {
            return (long)np.min(arr).GetItem(0);
        }
        internal ulong GetMin(ulong[] arr)
        {
            return (ulong)np.min(arr).GetItem(0);
        }
        internal int GetMin(int[] arr)
        {
            return (int)np.min(arr).GetItem(0);
        }
        internal uint GetMin(uint[] arr)
        {
            return (uint)np.min(arr).GetItem(0);
        }
        internal short GetMin(short[] arr)
        {
            return (short)np.min(arr).GetItem(0);
        }
        internal ushort GetMin(ushort[] arr)
        {
            return (ushort)np.min(arr).GetItem(0);
        }
        internal sbyte GetMin(sbyte[] arr)
        {
            return (sbyte)np.min(arr).GetItem(0);
        }
        internal byte GetMin(byte[] arr)
        {
            return (byte)np.min(arr).GetItem(0);
        }
        internal bool GetMin(bool[] arr)
        {
            return (bool)np.min(arr).GetItem(0);
        }

        internal double GetAverage(double[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(long[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(ulong[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(int[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(uint[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(short[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(ushort[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(sbyte[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(byte[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }
        internal double GetAverage(bool[] arr)
        {
            return (double)np.average(arr).GetItem(0);
        }

    }


    public static class Extensions
    {
        private static double Average(this sbyte[] array)
        {
            double total = 0;

            foreach (var a in array)
            {
                total += a;
            }

            return total / array.Length;
        }
        private static double Average(this byte[] array)
        {
            double total = 0;

            foreach (var a in array)
            {
                total += a;
            }

            return total / array.Length;
        }
        private static double Average(this Int16[] array)
        {
            double total = 0;

            foreach (var a in array)
            {
                total += a;
            }

            return total / array.Length;
        }
        private static double Average(this UInt16[] array)
        {
            double total = 0;

            foreach (var a in array)
            {
                total += a;
            }

            return total / array.Length;
        }
        private static double Average(this UInt32[] array)
        {
            double total = 0;

            foreach (var a in array)
            {
                total += a;
            }

            return total / array.Length;
        }
        private static double Average(this UInt64[] array)
        {
            double total = 0;

            foreach (var a in array)
            {
                total += a;
            }

            return total / array.Length;
        }

        public static sbyte[] FirstTen(this sbyte[] array)
        {
            sbyte[] firstTen = new sbyte[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static byte[] FirstTen(this byte[] array)
        {
            byte[] firstTen = new byte[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static Int16[] FirstTen(this Int16[] array)
        {
            Int16[] firstTen = new Int16[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static UInt16[] FirstTen(this UInt16[] array)
        {
            UInt16[] firstTen = new UInt16[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static Int32[] FirstTen(this Int32[] array)
        {
            Int32[] firstTen = new Int32[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static UInt32[] FirstTen(this UInt32[] array)
        {
            UInt32[] firstTen = new UInt32[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static Int64[] FirstTen(this Int64[] array)
        {
            Int64[] firstTen = new Int64[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static UInt64[] FirstTen(this UInt64[] array)
        {
            UInt64[] firstTen = new UInt64[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static float[] FirstTen(this float[] array)
        {
            float[] firstTen = new float[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }
        public static double[] FirstTen(this double[] array)
        {
            double[] firstTen = new double[Math.Min(10, array.Length)];
            Array.Copy(array, firstTen, firstTen.Length);
            return firstTen;
        }


    }
}

File: RandomnessLibraries/CPlusPlus/RndDistributionsUnitTest/TestBaseClass.cpp
#include "pch.h"
#include "TestBaseClass.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

void TestBaseClass::print(double d)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", d);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(double *d, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", d[i]);
		Logger::WriteMessage(buffer);
	}
}

void TestBaseClass::print(char b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(char *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}

void TestBaseClass::print(uchar b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(uchar *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}

void TestBaseClass::print(short b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(short *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}

void TestBaseClass::print(ushort b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(ushort *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}
void TestBaseClass::print(int b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(int *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}
void TestBaseClass::print(uint b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(uint *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}
void TestBaseClass::print(long b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(long *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}
void TestBaseClass::print(ulong b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(ulong *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}


void TestBaseClass::print(bool b)
{
	char buffer[256];

	sprintf_s(buffer, sizeof(buffer), "%d", b);
	Logger::WriteMessage(buffer);
}
void TestBaseClass::print(bool *b, int dsize)
{
	char buffer[256];

	for (int i = 0; i < dsize; i++)
	{
		sprintf_s(buffer, sizeof(buffer), "%d", b[i]);
		Logger::WriteMessage(buffer);
	}
}

///////////////////////////////////

double TestBaseClass::GetMax(double *arr, int size)
{
	double highNum = -DBL_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

double TestBaseClass::GetMin(double *arr, int size)
{
	double lowNum = DBL_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(double *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}

void TestBaseClass::FirstTen(double *arr, double *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}
//////////////////////////////


bool TestBaseClass::GetMax(bool *arr, int size)
{
	double highNum = true;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

bool TestBaseClass::GetMin(bool *arr, int size)
{
	double lowNum = false;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}
void TestBaseClass::FirstTen(bool *arr, bool *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}
////////////////////////////

char TestBaseClass::GetMax(char *arr, int size)
{
	char highNum = CHAR_MIN;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

char TestBaseClass::GetMin(char *arr, int size)
{
	char lowNum = CHAR_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(char *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(char *arr, char *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

uchar TestBaseClass::GetMax(uchar *arr, int size)
{
	uchar highNum = 0;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

uchar TestBaseClass::GetMin(uchar *arr, int size)
{
	uchar lowNum = UCHAR_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(uchar *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(uchar *arr, uchar *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

short TestBaseClass::GetMax(short *arr, int size)
{
	short highNum = SHRT_MIN;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

short TestBaseClass::GetMin(short *arr, int size)
{
	short lowNum = SHRT_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(short *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(short *arr, short *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

ushort TestBaseClass::GetMax(ushort *arr, int size)
{
	ushort highNum = 0;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

ushort TestBaseClass::GetMin(ushort *arr, int size)
{
	ushort lowNum = USHRT_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(ushort *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(ushort *arr, ushort *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

int TestBaseClass::GetMax(int *arr, int size)
{
	int highNum = INT_MIN;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

int TestBaseClass::GetMin(int *arr, int size)
{
	int lowNum = INT_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(int *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(int *arr, int *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

uint TestBaseClass::GetMax(uint *arr, int size)
{
	uint highNum = 0;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

uint TestBaseClass::GetMin(uint *arr, int size)
{
	uint lowNum = UINT_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(uint *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(uint *arr, uint *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

long TestBaseClass::GetMax(long *arr, int size)
{
	long highNum = LONG_MIN;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

long TestBaseClass::GetMin(long *arr, int size)
{
	long lowNum = LONG_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(long *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(long *arr, long *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

////////////////////////////

ulong TestBaseClass::GetMax(ulong *arr, int size)
{
	ulong highNum = 0;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] > highNum)
			highNum = arr[i];
	}
	return highNum;
}

ulong TestBaseClass::GetMin(ulong *arr, int size)
{
	ulong lowNum = ULLONG_MAX;
	int i;

	for (i = 0; i < size; i++)
	{
		if (arr[i] < lowNum)
			lowNum = arr[i];
	}
	return lowNum;
}

double TestBaseClass::GetAverage(ulong *arr, int size)
{
	double total = 0;

	for (int i = 0; i < size; i++)
	{
		total += arr[i];
	}

	return total / size;
}
void TestBaseClass::FirstTen(ulong *arr, ulong *firstTen)
{
	for (int i = 0; i < 10; i++)
	{
		firstTen[i] = arr[i];
	}
}

void TestBaseClass::AssertArray(bool *a, bool *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}
void TestBaseClass::AssertArray(char *a, char *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

void TestBaseClass::AssertArray(uchar *a, uchar *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

void TestBaseClass::AssertArray(short *a, short *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

//void TestBaseClass::AssertArray(ushort *a, ushort *b, int size)
//{
//	for (int i = 0; i < size; i++)
//	{
//		Assert::AreEqual(a[i], b[i]);
//	}
//}

void TestBaseClass::AssertArray(int *a, int *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

void TestBaseClass::AssertArray(uint *a, uint *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

void TestBaseClass::AssertArray(long *a, long *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

void TestBaseClass::AssertArray(ulong *a, ulong *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i]);
	}
}

void TestBaseClass::AssertArray(double *a, double *b, int size)
{
	for (int i = 0; i < size; i++)
	{
		Assert::AreEqual(a[i], b[i], doubleTolerance);
	}
}


int * TestBaseClass::arange_Int32(int start, int end)
{
	int bufLength = end - start;
	int * buffer = new int[bufLength];

	for (int i = 0; i < bufLength; i++)
	{
		buffer[i] = i + start;
	}

	return buffer;
}

double* TestBaseClass::arange_Double(int start, int end)
{
	int bufLength = end - start;
	double* buffer = new double[bufLength];

	for (int i = 0; i < bufLength; i++)
	{
		buffer[i] = (double)(i + start);
	}

	return buffer;
}

bool* TestBaseClass::CompareArray(long* arr, int arrlen, int c)
{
	bool* output = new bool[arrlen];

	for (int i = 0; i < arrlen; i++)
	{
		output[i] = arr[i] == c;
	}
	return output;
}


int TestBaseClass::CountTrues(bool* v, int vlength)
{
	int count = 0;

	for (int i = 0; i < vlength; i++)
	{
		count += v[i] == true ? 1 : 0;
	}

	return count;
}



